runners/google-cloud-dataflow-java/worker/src/test/java/org/apache/beam/runners/dataflow/worker/windmill/client/grpc/stubs/ChannelCacheTest.java

```diff
@@ -132,12 +136,80 @@ public void testClear() throws InterruptedException {
     CountDownLatch notifyWhenChannelClosed = new CountDownLatch(1);
     cache =
         ChannelCache.forTesting(
-            ignored -> newChannel(channelName), notifyWhenChannelClosed::countDown);
+            (a, b) -> newChannel(channelName), notifyWhenChannelClosed::countDown);
     WindmillServiceAddress someAddress = mock(WindmillServiceAddress.class);
     ManagedChannel cachedChannel = cache.get(someAddress);
     cache.clear();
     notifyWhenChannelClosed.await();
     assertTrue(cache.isEmpty());
     assertTrue(cachedChannel.isShutdown());
   }
+
+  @Test
+  public void testConsumeFlowControlSettings() throws InterruptedException {
+    String channelName = "channel";
+    CountDownLatch notifyWhenChannelClosed = new CountDownLatch(1);
+    AtomicInteger newChannelsCreated = new AtomicInteger();
+    cache =
+        ChannelCache.forTesting(
+            (a, b) -> {
```

Comment:
seems like you should keep track of what the settings passed in are, to make sure the plumbing is right. for example could add the settings to some concurrentqueue and then validate below